src/pm/hydra/pm/pmiserv/pmip_utils.c

```diff
@@ -277,13 +277,9 @@ static HYD_status global_env_fn(char *arg, char ***argv)
             str[strlen(str) - 1] = 0;
         }
 
-        if (!strcmp(arg, "global-inherited-env")) {
+        if (!strcmp(arg, "global-inherited-env"))
             HYDU_append_env_str_to_list(str, &HYD_pmcd_pmip.user_global.global_env.inherited);
-            /* Make sure to let proxy aware of HYDRA related variables */
-            if (!strncmp(str, "HYDRA_", 6)) {
-                MPL_putenv(str);
-            }
```

Comment:
Will it work if we do:
```
if (strcmp(str, "HYDRA_TOPO_DEBUG") == 0) {
    HYDT_topo_info.debug = 1;
}
```
?

Reply:
We'd also have to compare the value and make sure it's set to 1, or some equivalent allowed by MPL_env2bool.
```
    val_ptr = getenv(envName);
    if (val_ptr) {
        if (strcmp(val_ptr, "YES") == 0 ||
            strcmp(val_ptr, "yes") == 0 ||
            strcmp(val_ptr, "TRUE") == 0 ||
            strcmp(val_ptr, "true") == 0 ||
            strcmp(val_ptr, "ON") == 0 || strcmp(val_ptr, "on") == 0 || strcmp(val_ptr, "1") == 0) {
```

Reply:
I see. So alternatively we just need `MPL_str2bool`, identical to `MPL_env2bool` but with a `NAME=val` string? I think that is better than adding code in all other 5 files. It is at least more scalable, right? Say if we have another debug option, we can just add another `MPL_str2bool("HYDRA_DEBUG_ANOTHER", &HYDT_some_dbg_flag);` rather than adding changes in 5 files again.

Practically, this is a debug flag, so I don't ever see the need to worry about user passing `HYDRA_TOPO_DEBUG=NO` or equivalent. But for completeness, I think adding `MPL_str2bool` is better.

Reply:
Actually, we may just need `MPL_str2bool(strval)` to return `1` or `0`, and use it like this:
```
if (strcmp(str, "HYDRA_TOPO_DEBUG=") == 0) {
    HYDT_some_dbg_flag = MPL_str2bool(str + 17);
}
```
`MPL_env2bool` can be refactored to use `MPL_str2bool` so we don't duplicate the code.

Reply:
Are you proposing to eliminate all proxy launch arguments?

Reply:
I was not. But now you mentioned, it is not a bad idea. `mpiexec` encodes the launch argument one way or another, unifying them is not a bad idea, right?

Now that you mentioned, I recall @pavanbalaji made a similar comment back then. In fact that is the story behind this issue: https://github.com/pmodels/mpich/issues/3434

Reply:
I would like to fix this *user-visible* bug before undertaking any large refactoring efforts.

Reply:
I agree, therefore I emphasize that I was not "proposing to eliminate all proxy launch arguments" at this time. This comment [https://github.com/pmodels/mpich/pull/4604#discussion_r434948373] was to suggest one way of fixing this bug.

Reply:
I would argue for launch arguments over psuedo environment variables passed to the proxy. Less fragile.

Reply:
Do you realize that both are passed to proxy via command line? Why would one be more fragile than the other? Anyway, I agree it is plausible to drop the environment at proxy and consolidate them to command line especially since the other 4 parameters are in the command line options. For consistency, we should get rid of the environment checking of the other 4 topo parameters too -- refer to my other comments below.